### PR TITLE
Change the default verbose level to INFO

### PR DESCRIPTION
**Pull request by shihchengli**
## Description
Currently, only ERROR level messages are printed. Setting the default to INFO can ensure that more useful information is logged.

## Checklist
- [ ] linted with flake8?
- [ ] (if appropriate) unit tests added?

### CODE REVIEW

**Comment by JacksonBurns** [comment]
I would personally prefer to leave our current default - I think it is also similar to what other packages do. If this PR is related to the linked issue, perhaps we just add a line in the docs (or raise a warning) suggesting that users turn on info level debugging when the situation calls for it?

**Comment by shihchengli** [comment]
The current issue is that the entire test results are not printed out unless the verbose level is changed. We also don't have many different levels of messages that can be logged in the CLI. For example, at the INFO level, the training would provide (1) the location and scale of scalers for extra atom, bond, and molecular descriptors if provided, (2) the path to the loaded data, (3) split sizes, (4) the default loss function if the loss function is not specified, (5) model architecture, (6) path to save the best model, (7) entire test results, and (8) the means and standard deviations of targets. I think most of these are quite common, and users would expect us to provide them in the log files, especially the overall test results. We only have two DEBUG level messages in the codebase, which are the evaluation metric used to monitor the model training in train.py and hpopt.py. I also noticed we don't have any ERROR level messages in the CLI, and for warning messages, we always use `warnings.warn` instead of `logger.warn()`. This makes me wonder about the necessity of setting different verbose levels and feel like we can always print out the messages we think are useful. Maybe we will have more comprehensive messages added in the future (e.g., #882), so it would still be good to keep different levels of messages. However, I think keeping those essential messages printed out or saved in another file is more user-friendly.

**Comment by KnathanM** [comment]
In line with Shih-Cheng's comments, `grep -Iirn "logger" chemprop/*` shows me that we only ever use `logger.debug` and `logger.info`. 

But the logger is also used by other modules like fsspec which writes lines like:
`2024-07-03T16:24:36 - DEBUG:fsspec.local - open file: /.../model_0/checkpoints/best-epoch=4-val_loss=0.75.ckpt`. 
We wouldn't want those always on by default. 

The reason we use `warnings.warn` is that those calls are not in the CLI so they don't use logger. 

I think the solution is to think of `logger.info` as more for FYI's instead of for important information. I'd be okay with printing at least some of the things you mentioned to stdout so that it goes where lightning puts some of its output instead of putting it with the logging output (which may go to a file). Maybe we go through the list of things that we write to `logger.info` in our next dev meeting and decide which of them we want to always write to stdout (probably by using print instead of logger).

**Comment by davidegraff** [comment]
A quick response to @shihchengli 's point: The lack of variety in utilized logging levels is not a reason to alter the default log level. Rather, it's an indictment of how we've used logging levels. I agree with your general point though, and we should probably think about (1) changing the levels of some calls to be in line with other packages and (2) adding more varied log statements. Logging does clog up the source, but it ultimately helps downstream users both at the library and CLI levels. We should opt for _more_ information rather than less.

**Comment by KnathanM** [comment]
I now think that we should change our default verbose level to INFO.

Lightning uses INFO by [default](https://github.com/Lightning-AI/pytorch-lightning/blob/f5d82504da252a255df268bd2bb99bf3f2886d27/src/lightning/__init__.py#L7). Most of the messages lightning prints out go to the INFO channel like these [lines](https://github.com/Lightning-AI/pytorch-lightning/blob/f5d82504da252a255df268bd2bb99bf3f2886d27/src/lightning/pytorch/trainer/setup.py#L144) telling you if the GPU is being used.

Lightning once had a [discussion](https://github.com/Lightning-AI/pytorch-lightning/issues/710) about whether to use INFO or WARNING as the default level. They settled on keeping INFO as the default, but being more careful to only set the level for the lightning module logger so as to not affect client logging code. We could do this as well. 

If we do make this change, maybe we keep `-v` in the CLI to mean set the logging level to DEBUG and add `-q` to set the logging level to WARNING. Not sure if we want `-qq` to mean set the level to ERROR.

**Comment by davidegraff** [comment]
I like that suggestion!

**Comment by shihchengli** [comment]
@KnathanM Thanks for sharing this information. I like this suggestion.

**Review by KnathanM — APPROVED** [review]
